package.json

```diff
@@ -185,43 +187,40 @@
     "husky": "^0.11.9",
     "ignore-styles": "^5.0.1",
     "immer": "^7.0.15",
-    "jsdom": "^21.1.2",
+    "jsdom": "^24.1.1",
     "jsdom-global": "^3.0.2",
     "less": "^3.11.1",
     "less-loader": "^11.1.3",
     "mini-css-extract-plugin": "^2.7.6",
-    "mocha": "^10.2.0",
+    "mocha": "^10.7.0",
     "npm-run-all": "^4.1.5",
     "nyc": "^14.1.1",
     "path-browserify": "^1.0.1",
     "pirates": "^4.0.6",
     "prop-types": "^15.6.0",
     "proxyquire": "^1.7.11",
     "raf": "^3.4.0",
-    "rc-resize-observer": "^1.0.0",
-    "react": "^16.13.1",
+    "react": "^18.3.1",
     "react-checkbox-tree": "^1.8.0",
     "react-clipboard.js": "^1.1.2",
-    "react-dom": "^16.13.1",
+    "react-dom": "^18.3.1",
     "react-hotkeys": "^2.0.0",
-    "react-html-parser": "^2.0.2",
     "react-markdown": "^6.0.3",
     "react-redux": "^7.2.1",
-    "react-virtualized": "^9.22.3",
-    "reapop": "4.0.5",
+    "react-virtualized": "^9.22.5",
+    "reapop": "^4.2.2",
     "reduce-reducers": "^1.0.4",
     "redux": "^4.0.5",
     "redux-logger": "^3.0.6",
     "redux-mock-store": "^1.5.1",
     "redux-thunk": "^2.2.0",
     "reselect": "^4.1.6",
     "rimraf": "^2.6.3",
-    "semantic-ui-react": "^1.2.1",
+    "semantic-ui-react": "^3.0.0-beta.2",
```

Comment:
Why `beta`?
Other versions does not support react@18?

Reply:
Only v3 supports native ref forwarding: https://github.com/Semantic-Org/Semantic-UI-React/issues/4438